components/skus/browser/sdk_controller.cc

```diff
@@ -0,0 +1,153 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/components/skus/browser/sdk_controller.h"
+
+#include <utility>
+
+#include "base/json/json_reader.h"
+#include "base/task/post_task.h"
+#include "base/threading/sequenced_task_runner_handle.h"
+#include "brave/components/skus/browser/pref_names.h"
+#include "brave/components/skus/browser/rs/cxx/src/lib.rs.h"
+#include "brave/components/skus/browser/rs/cxx/src/shim.h"
+#include "brave/components/skus/browser/skus_sdk_context_impl.h"
+#include "brave/components/skus/browser/skus_utils.h"
+#include "components/prefs/pref_service.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+
+namespace {
+
+void OnRefreshOrder(skus::RefreshOrderCallbackState* callback_state,
+                    skus::SkusResult result,
+                    rust::cxxbridge1::Str order) {
+  std::string order_str = static_cast<std::string>(order);
+  if (callback_state->cb) {
+    std::move(callback_state->cb).Run(order_str);
+  }
+  delete callback_state;
+}
+
+void OnFetchOrderCredentials(
+    skus::FetchOrderCredentialsCallbackState* callback_state,
+    skus::SkusResult result) {
+  if (callback_state->cb) {
+    std::move(callback_state->cb).Run("");
+  }
+  delete callback_state;
+}
+
+void OnPrepareCredentialsPresentation(
+    skus::PrepareCredentialsPresentationCallbackState* callback_state,
+    skus::SkusResult result,
+    rust::cxxbridge1::Str presentation) {
+  if (callback_state->cb) {
+    std::move(callback_state->cb).Run(static_cast<std::string>(presentation));
+  }
+  delete callback_state;
+}
+
+void OnCredentialSummary(skus::CredentialSummaryCallbackState* callback_state,
+                         skus::SkusResult result,
+                         rust::cxxbridge1::Str summary) {
+  std::string summary_string = static_cast<std::string>(summary);
+  base::JSONReader::ValueWithError value_with_error =
+      base::JSONReader::ReadAndReturnValueWithError(
+          summary_string, base::JSONParserOptions::JSON_PARSE_RFC);
+  absl::optional<base::Value>& records_v = value_with_error.value;
+
+  if (records_v && callback_state->prefs) {
+    if (callback_state->domain == "vpn.brave.com" ||
+        callback_state->domain == "vpn.bravesoftware.com" ||
+        callback_state->domain == "vpn.brave.software") {
+      const base::Value* active = records_v->FindKey("active");
+      if (active) {
+        bool has_credential = active && active->is_bool() && active->GetBool();
+        callback_state->prefs->SetBoolean(skus::prefs::kSkusVPNHasCredential,
+                                          has_credential);
+      }
+    }
+  }
+
+  if (callback_state->cb) {
+    std::move(callback_state->cb).Run(summary_string);
+  }
+  delete callback_state;
+}
+
+}  // namespace
+
+namespace skus {
+
+SdkController::SdkController(
```

Comment:
I think this naming change is worse. The `Controller` naming convention in wallet is a throwback to a time before they were keyed services and we shouldn't be following that convention

Reply:
Fixed in https://github.com/brave/brave-core/pull/10847/commits/b4c7929ed36eec752a173b6a5f60e2ec8fd8f33b